xchange-ftx/src/main/java/org/knowm/xchange/ftx/service/FtxLendingServiceRaw.java

```diff
@@ -0,0 +1,277 @@
+      Optional<FtxLendingInfoDto> op = infos.stream()
+          .filter(lendingInfo -> lendingInfo.getCoin().equalsIgnoreCase(coin))
+          .findFirst();
+
+      if (!op.isPresent()) throw new FtxLendingServiceException("Cant lend all, infos don't exist for coin: " + coin);
+
+      double sizeToLend = FtxAdapters.lendingRounding(new BigDecimal(op.get().getLendable())).doubleValue();
+      if (Double.compare(sizeToLend, op.get().getOffered()) == 0)
+        list.add(new FtxLendDataDto(coin, op.get().getLocked(), sizeToLend, sizeToLend, rate));
+      else {
+        try {
+          ftx.submitLendingOffer(
+              exchange.getExchangeSpecification().getApiKey(),
+              exchange.getNonceFactory().createValue(),
+              signatureCreator,
+              subaccount,
+              new FtxSubmitLendingOfferParams(coin, sizeToLend, rate)
+          );
+          list.add(new FtxLendDataDto(coin, op.get().getLocked(), op.get().getOffered(), sizeToLend, rate));
+        } catch (IOException e) {
+          throw new FtxLendingServiceException("Can't lend all for subaccount: " + subaccount + ", coin: " + coin + ", rate: " + rate);
+        }
+      }
+    });
+    return new ArrayList<>();
+  }
+
+  public FtxLendDataDto lend(String subaccount, String coin, double size, double rate) {
+    Objects.requireNonNull(coin);
+    if (StringUtils.isNotBlank(coin))
+      throw new FtxLendingServiceException("Coin are blank or empty");
+    if (rate < 0)
+      throw new FtxLendingServiceException("Rate must to be >= 0, subaccount: " + subaccount + ", coin: " + coin);
+    if (size < 0)
+      throw new FtxLendingServiceException("Size must to be >= 0, subaccount: " + subaccount + ", coin: " + coin + ", rate: " + rate);
+
+    try {
+      FtxLendingInfoDto info = info(subaccount, coin);
+      double sizeToLend = FtxAdapters.lendingRounding(new BigDecimal(size)).doubleValue();
+
+      if (Double.compare(sizeToLend, info.getLendable()) == 1) {
+        throw new FtxLendingServiceException("Cant't lend > to lendable, subaccount: " + subaccount + ", coin: " + coin + ", size: " + size + ", sizeToLend: " + sizeToLend + ", rate: " + rate);
+      }
+      ftx.submitLendingOffer(
+          exchange.getExchangeSpecification().getApiKey(),
+          exchange.getNonceFactory().createValue(),
+          signatureCreator,
+          subaccount,
+          new FtxSubmitLendingOfferParams(coin, FtxAdapters.lendingRounding(new BigDecimal(sizeToLend)).doubleValue(), rate)
+      );
+      return new FtxLendDataDto(coin, info.getLocked(), info.getOffered(), sizeToLend, rate);
+    } catch (IOException e) {
+      throw new FtxLendingServiceException("Can't lend subaccount: " + subaccount + ", coin: " + coin + ", size: " + size + ", rate: " + rate, e);
+    }
+  }
+
+  public FtxLendDataDto lend(String subaccount, String coin, FtxLendingInfoFunction functionSize, FtxLendingInfoFunction functionRate) {
```

Comment:
this function and the function directly above are extremely similar, can you compute the size & rate in this function and then call the above function with the results so as to remove duplicate code

Reply:
ok np

Reply:
you tell me remove this method

Reply:
try and collapse the logic of these methods together and share the logic that is the same

Reply:
I remove unnecessary methods